addons/packages/vsphere-cpi/1.22.5/test/vsphere_cpi_test.go

```diff
@@ -0,0 +1,297 @@
+			hash, exist := daemonSet.Spec.Template.Annotations["vsphere-cpi/data-values-hash"]
+			Expect(exist).To(BeTrue())
+			Expect(hash).Should(HavePrefix("h-"))
+		})
+	})
+
+	Context("IPFamilies", func() {
+		Context("unset", func() {
+			BeforeEach(func() {
+				values = `#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+vsphereCPI:
+  server: fake-server.com
+  datacenter: dc0
+  username: my-user
+  password: my-password-2
+  insecureFlag: True`
+			})
+
+			It("defaults to ipv4", func() {
+				Expect(yttRenderErr).NotTo(HaveOccurred())
+				Expect(configuredIPFamily(output)).To(ConsistOf("ipv4"))
+			})
+		})
+		Context("IPv4", func() {
+			BeforeEach(func() {
+				values = `#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+vsphereCPI:
+  server: fake-server.com
+  datacenter: dc0
+  username: my-user
+  password: my-password
+  insecureFlag: True
+  ipFamily: ipv4`
+			})
+
+			It("configures ipv4", func() {
+				Expect(yttRenderErr).NotTo(HaveOccurred())
+				Expect(configuredIPFamily(output)).To(ConsistOf("ipv4"))
+			})
+		})
+		Context("IPv6", func() {
+			BeforeEach(func() {
+				values = `#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+vsphereCPI:
+  server: fake-server.com
+  datacenter: dc0
+  username: my-user
+  password: my-password
+  insecureFlag: True
+  ipFamily: ipv6`
+			})
+
+			It("configures ipv6", func() {
+				Expect(yttRenderErr).NotTo(HaveOccurred())
+				Expect(configuredIPFamily(output)).To(ConsistOf("ipv6"))
+			})
+		})
+		Context("IPv4,IPv6", func() {
+			BeforeEach(func() {
+				values = `#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+vsphereCPI:
+  server: fake-server.com
+  datacenter: dc0
+  username: my-user
+  password: my-password
+  insecureFlag: True
+  ipFamily: ipv4,ipv6`
+			})
+
+			It("configures ipv4,ipv6", func() {
+				Expect(yttRenderErr).NotTo(HaveOccurred())
+				Expect(configuredIPFamily(output)).To(Equal([]string{"ipv4", "ipv6"}))
+			})
+		})
+		Context("IPv6,IPv4", func() {
+			BeforeEach(func() {
+				values = `#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+vsphereCPI:
+  server: fake-server.com
+  datacenter: dc0
+  username: my-user
+  password: my-password
+  insecureFlag: True
+  ipFamily: ipv6,ipv4`
+			})
+
+			It("configures ipv6,ipv4", func() {
+				Expect(yttRenderErr).NotTo(HaveOccurred())
+				Expect(configuredIPFamily(output)).To(Equal([]string{"ipv6", "ipv4"}))
+			})
+		})
+		Context("Garbage", func() {
+			BeforeEach(func() {
+				values = `#@data/values
+#@overlay/match-child-defaults missing_ok=True
+---
+vsphereCPI:
+  server: fake-server.com
+  datacenter: dc0
+  username: my-user
+  password: my-password
+  insecureFlag: True
+  ipFamily: ipv5`
+			})
+
+			It("errors when rendering", func() {
+				Expect(yttRenderErr).To(MatchError(ContainSubstring("vsphereCPI ipFamily should be one of \"ipv4\", \"ipv6\", \"ipv4,ipv6\", or \"ipv6,ipv4\" if provided")))
+			})
+		})
```

Comment:
We should be able to add the Node IP Selectors tests from 1.23.0 to cover the new exclude behavior https://github.com/vmware-tanzu/community-edition/blob/5aaf4df530fce01545f823492e760229bbbbf644/addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go#L348-L471